domain/src/test/java/org/oppia/android/domain/profile/ProfileManagementControllerTest.kt

```diff
@@ -1278,6 +1279,126 @@ class ProfileManagementControllerTest {
+    assertThat(profile.isAdmin).isEqualTo(true)
+    assertThat(profile.avatar.avatarImageUri).isEqualTo("")
+    assertThat(profile.avatar.avatarColorRgb).isEqualTo(-1)
+  }
+
+  @Test
+  fun testUpdateProfile_updateMultipleFields_invalidName_checkUpdateFailed() {
+    setUpTestApplicationComponent()
+    profileTestHelper.createDefaultProfile()
+
+    val updateProvider = profileManagementController.updateNewProfileDetails(
+      PROFILE_ID_0,
+      ProfileType.SOLE_LEARNER,
+      null,
+      -1,
+      "John123",
+      isAdmin = true
+    )
+    val failure = monitorFactory.waitForNextFailureResult(updateProvider)
+
+    assertThat(failure).hasMessageThat().contains("John123 does not contain only letters")
+  }
+
+  @Test
+  fun testUpdateProfile_updateMultipleFields_nullAvatarUri_setsAvatarColorSuccessfully() {
+    setUpTestApplicationComponent()
+    profileTestHelper.createDefaultProfile()
+
+    val updateProvider = profileManagementController.updateNewProfileDetails(
+      PROFILE_ID_0,
+      ProfileType.SOLE_LEARNER,
+      null,
+      -11235672,
+      "John",
+      isAdmin = true
+    )
+    monitorFactory.waitForNextSuccessfulResult(updateProvider)
+
+    val profileProvider = profileManagementController.getProfile(PROFILE_ID_0)
+    val profile = monitorFactory.waitForNextSuccessfulResult(profileProvider)
+
+    assertThat(profile.avatar.avatarImageUri).isEqualTo("")
+    assertThat(profile.avatar.avatarColorRgb).isEqualTo(-11235672)
+  }
+
+  @Test
+  fun testUpdateProfile_updateMultipleFields_invalidProfileId_checkUpdateFailed() {
+    setUpTestApplicationComponent()
+    profileTestHelper.createDefaultProfile()
+
+    val updateProvider = profileManagementController.updateNewProfileDetails(
+      PROFILE_ID_3,
+      ProfileType.SOLE_LEARNER,
+      null,
+      -1,
+      "John",
+      isAdmin = true
+    )
+    val failure = monitorFactory.waitForNextFailureResult(updateProvider)
+
+    assertThat(failure).hasMessageThat()
+      .contains("ProfileId ${PROFILE_ID_3?.internalId} does not match an existing Profile")
+  }
+
+  @Test
+  fun testUpdateProfile_updateProfileType_existingAdminProfile_checkUpdateSucceeded() {
+    setUpTestApplicationComponent()
+    profileTestHelper.addOnlyAdminProfile()
+
+    val updateProvider = profileManagementController.updateProfileType(
+      PROFILE_ID_0,
+      ProfileType.SUPERVISOR
+    )
+    monitorFactory.waitForNextSuccessfulResult(updateProvider)
+  }
+
+  @Test
+  fun testUpdateProfile_updateProfileType_existingNonAdminProfile_checkUpdateSucceeded() {
+    setUpTestApplicationComponent()
+    addNonAdminProfileAndWait(name = "Rajat", pin = "01234")
+
+    val updateProvider = profileManagementController.updateProfileType(
+      PROFILE_ID_0,
+      ProfileType.ADDITIONAL_LEARNER
+    )
+    monitorFactory.waitForNextSuccessfulResult(updateProvider)
+  }
+
+  @Test
+  fun testUpdateProfile_updateProfileType_newDefaultProfile_checkUpdateSucceeded() {
```

Comment:
These new tests should also verify that the corresponding fields of the profile were successfully updated.

Reply:
I don't understand. I have tests like this for the api that only updates one field, and a seperate api+corresponding tests that update multiple tests

Reply:
Ah I see. So the outcome part of the tests is confusing: it should probably be clear on what is 'successful'. It seems that some tests are verifying that the operation doesn't fail (such as this one), and others are verifying that the actual profile was changed on disk (``testUpdateProfile_updateMultipleFields_checkUpdateIsSuccessful``). I suggest renaming these tests such that their expected outcomes are clearer on what, exactly, should happen rather than using 'succeeded/successful' since those terms are ambiguous here.

Reply:
I have renamed the updateProfileType tests, and also added assertions for the corresponding fields in the tests for the multiple fields api.